tests/e2e/functional-nc-e2e/numaflow-controller_test.go

```diff
@@ -0,0 +1,249 @@
+			UDSource: &numaflowv1.UDSource{
+				Container: &numaflowv1.Container{
+					Image: "quay.io/numaio/numaflow-go/source-simple-source:stable",
+				},
+			},
+		},
+		Sink: &numaflowv1.Sink{
+			AbstractSink: numaflowv1.AbstractSink{
+				UDSink: &numaflowv1.UDSink{
+					Container: &numaflowv1.Container{
+						Image: "quay.io/numaio/numaflow-go/sink-log:stable",
+					},
+				},
+			},
+		},
+	}
+)
+
+func TestFunctionalE2E(t *testing.T) {
+	RegisterFailHandler(Fail)
+
+	BeforeSuite(func() {
+		BeforeSuiteSetup()
+	})
+
+	RunSpecs(t, "Functional E2E Suite")
+}
+
+var _ = Describe("Functional e2e:", Serial, func() {
+
+	It("Should create the NumaflowControllerRollout if it doesn't exist", func() {
+		CreateNumaflowControllerRollout(InitialNumaflowControllerVersion)
+	})
+
+	It("Should create the ISBServiceRollout if it doesn't exist", func() {
+		CreateISBServiceRollout(isbServiceRolloutName, isbServiceSpec)
+	})
+
+	It("Should create the PipelineRollout if it does not exist", func() {
+		CreatePipelineRollout(pipelineRolloutName, Namespace, initialPipelineSpec, false)
+	})
+
+	It("Should create the MonoVertexRollout if it does not exist", func() {
+		CreateMonoVertexRollout(monoVertexRolloutName, Namespace, initialMonoVertexSpec)
+	})
+
+	It("Should update the child NumaflowController if the NumaflowControllerRollout is updated", func() {
+		UpdateNumaflowControllerRollout(InitialNumaflowControllerVersion, UpdatedNumaflowControllerVersion, []PipelineRolloutInfo{{PipelineRolloutName: pipelineRolloutName}}, true)
+		VerifyISBSvcReady(Namespace, isbServiceRolloutName, 3)
+		VerifyPipelineRolloutHealthy(pipelineRolloutName)
+		VerifyPipelineRunning(Namespace, pipelineRolloutName)
+		VerifyMonoVertexRolloutReady(monoVertexRolloutName)
+		Expect(VerifyMonoVertexReady(Namespace, monoVertexRolloutName)).ShouldNot(HaveOccurred())
+	})
+
+	if UpgradeStrategy == config.PPNDStrategyID {
+		It("Should pause the Pipeline if user requests it", func() {
+
+			By("setting desiredPhase=Paused")
+			currentPipelineSpec := updatedPipelineSpec
+			currentPipelineSpec.Lifecycle.DesiredPhase = numaflowv1.PipelinePhasePaused
+
+			UpdatePipelineRollout(pipelineRolloutName, currentPipelineSpec, numaflowv1.PipelinePhasePaused, func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
+				return retrievedPipelineSpec.Lifecycle.DesiredPhase == numaflowv1.PipelinePhasePaused
+			}, false)
+
+			VerifyPipelineStaysPaused(pipelineRolloutName)
+		})
+
+		It("Should resume the Pipeline if user requests it", func() {
+
+			By("setting desiredPhase=Running")
+			currentPipelineSpec := updatedPipelineSpec
+			currentPipelineSpec.Lifecycle.DesiredPhase = numaflowv1.PipelinePhaseRunning
+
+			UpdatePipelineRollout(pipelineRolloutName, currentPipelineSpec, numaflowv1.PipelinePhaseRunning, func(retrievedPipelineSpec numaflowv1.PipelineSpec) bool {
+				return retrievedPipelineSpec.Lifecycle.DesiredPhase == numaflowv1.PipelinePhaseRunning
+			}, false)
+		})
+	}
```

Comment:
I figured this would be in the pipeline test - any reason why it's in the numaflow controller test?

Reply:
[Here](https://github.com/numaproj/numaplane/issues/581#issuecomment-2682597715) it's mentioned as "Pipeline pausing for PPND" in NC test check?

Reply:
Sorry for not clarifying, but that's just related to the pausing which occurs when updating the Numaflow Controller itself. So, we can take this section out.

Reply:
Done